src/cephadm/cephadm

```diff
@@ -5212,6 +5222,185 @@ def command_gather_facts():
 ##################################
 
 
+class CephadmDaemonHandler(BaseHTTPRequestHandler):
+    def do_GET(self):
+        # return the current state of the cache
+        self.send_response(200)
+        self.send_header('Content-type','application/json')
+        self.end_headers()
+        self.wfile.write(bytes(json.dumps(cephadm_cache), 'ascii'))
+    
+    def log_message(self, format, *args):
+        rqst = " ".join(args)
+        logger.info(f"client:{self.address_string()} [{self.log_date_time_string()}] {rqst}")
+
+
+class CephadmHTTPServer(ThreadingMixIn, HTTPServer):
+    allow_reuse_address = True
+    daemon_threads = True
+
+
+def cephadmd_install():
+    # open firewall port
+    # copy self to /usr/bin somewhere
+    # create the systemd unit file
+    # enable the systemd unit file
+    # start the systemd unit
+    pass
```

Comment:
There are two ways to deploy this:

1. can we add this to `command_deploy` and thus only have a single way to deploy daemons. The we could think of this daemon as if it is just another type of daemon. 
* Pro: might be a logical step in the direction of adding keepalived (or something similar)
* Pro: benefit from `ServiceSpec`
* Pro: benefit from the scheduler
* Con: The daemon is not really similar to the other services as it is not a container

2. Make it a dedicated command and treat the daemons aas something special
* Pro: similar to the deployment of `/etc/ceph/ceph.conf`
* Con: No super easy integration into mgr/cephadm
* Con: Hard to add cephx keys for the daemons.

Reply:
this was from a prior commit - current code uses deploy and rm-daemon